devito/function.py

```diff
@@ -740,6 +747,26 @@ def dt2(self):
 
         return self.diff(_t, _t).as_finite_difference(indt)
 
+    @property
+    def _time_size(self):
+        return self.shape_allocated[self._time_position]
+
+    @property
+    def _time_buffering(self):
+        return self.save is not int
+
+    @property
+    def _time_buffering_default(self):
+        return self._time_buffering and self.save != Buffer
+
+    def _arg_check(self, args, intervals):
+        super(TimeFunction, self)._arg_check(args, intervals)
+        key = args[self.name]
+        if self._time_buffering and self._time_size != key.shape[self._time_position]:
+            raise InvalidArgument("Expected `time_size=%d` for runtime "
+                                  "value `%s`, found `%d` instead"
+                                  % (self._time_size, self.name, key._time_size))
```

Comment:
Not for this PR, but these multiline error message do not process the `%` arguments correctly in a lot of cases

Reply:
I didn't know that but yes, we should definitely fix this

Reply:
Actually, I just hit a problem with this anyway:
`key._time_size` => `key.shape[self._time_position]`?

Reply:
isn't it the same thing?

https://github.com/opesci/devito/blob/timeslot-avoidance/devito/function.py#L751

Reply:
It just doesn't match the comparison immediately above it, is all.
I don't know what the underlying error was in my case - it failed saying `key._time_size` was invalid before it could raise that exception.

Reply:
`key` was a `Data`, which appears to have been put there by `TensorFunction._arg_defaults`
and `Data` has no `_time_size`.